### PR TITLE
Tools: enforce astyle formatting in AP_DDS

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

File: Tools/scripts/run_astyle.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+    def run(self):
+        for d in self.directories_to_check:
+            self.find_files_in_directory(d)
+        self.check()
+        return self.retcode
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description='Check all Python files for flake8 cleanliness')
```
Comment (Ryanf55) [inline Tools/scripts/run_astyle.py:65]: Update to `astyle` cleanliness

File: Tools/scripts/run_astyle.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        ]
+        self.files_to_check = []
+
+    def progress(self, string):
+        print("****** %s" % (string,))
+
+    def check(self):
+        '''run astyle on all files in self.files_to_check'''
+        for path in self.files_to_check:
+            self.progress("Checking (%s)" % path)
```
Comment (Ryanf55) [inline Tools/scripts/run_astyle.py:32]: Now that we have f-strings, want to use them?

File: Tools/scripts/run_astyle.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env python
```
Comment (Ryanf55) [inline Tools/scripts/run_astyle.py:1]: python3?

File: Tools/scripts/run_astyle.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+            self.retcode = 1
+        if "Formatted" in ret.stdout:
+            self.progress("Files needing formatting found")
+            self.retcode = 1
+
+    def find_files_in_directory(self, d):
+        '''append filepaths of .c and .h files under d to self.files_to_check'''
+        for (dirpath, dirnames, filenames) in os.walk(d):
+            for filename in filenames:
+                if os.path.splitext(filename)[1] not in [".cpp", ".c", ".h"]:
```
Comment (Ryanf55) [inline Tools/scripts/run_astyle.py:52]: Can you define this list at the top of the file for clarity?

Comment (Ryanf55) [inline Tools/scripts/run_astyle.py:52]: Do you want to use pathlib instead?

File: Tools/scripts/run_astyle.py (hunk 1 above)
Comment (peterbarker) [reply Tools/scripts/run_astyle.py]: Fixed.

File: Tools/scripts/run_astyle.py (hunk 3 above)
Comment (peterbarker) [reply Tools/scripts/run_astyle.py]: Fixed

File: Tools/scripts/run_astyle.py (hunk 4 above)
Comment (peterbarker) [reply Tools/scripts/run_astyle.py]: Pretty sure that just obfuscates things.

Comment (peterbarker) [reply Tools/scripts/run_astyle.py]: Done.  I think I'm going to like our new Python3 thing.

File: Tools/scripts/run_astyle.py (hunk 2 above)
Comment (peterbarker) [reply Tools/scripts/run_astyle.py]: Next time :-)

File: Tools/scripts/run_astyle.py [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    def run(self):
+        for d in self.directories_to_check:
+            self.files_to_check.extend(list(pathlib.Path(d).glob("*")))
+        self.files_to_check = list(filter(lambda x : x.suffix in [".c", ".h", ".cpp"], self.files_to_check))
+        self.check()
+        return self.retcode
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description='Check all Python files for astyle cleanliness')
```
Comment (Ryanf55) [inline Tools/scripts/run_astyle.py:59]: Right now, there is no feedback when you run this on that it was successful. Can you add a `-v` option that prints the list of files, organized by library, that are being formatted? The behavior I observe is that the command only shows files if there is an error in one of them.

Example:
```
libraries/AP_DDS
    - AP_DDS_Client.h
    - AP_DDS_Generic_Fn_T.h  - 
    - AP_DDS_Topic_Table.h 
 AP_LIZ2
     - myfile.txt

**Review by Ryanf55 — APPROVED** [review]
Minor feedback, but approved anyways if you can't address it, since it meets the original needs.

**Pull request merged**